corehq/apps/reports/views.py

```diff
@@ -1242,6 +1244,12 @@ def dispatch(self, request, *args, **kwargs):
                           "Sorry, we couldn't find that case. If you think this "
                           "is a mistake please report an issue.")
             return HttpResponseRedirect(CaseListReport.get_url(domain=self.domain))
+        if not user_can_access_location_id(
+            self.domain,
+            request.couch_user,
+            self.case_instance.owner_id
+        ):
```

Comment:
I think this is overly restrictive - these users should also be able to access cases owned by users at accessible locations.  That is, I think you'd need to figure out if the case is owned by a user or a location.
The `CaseInfo` class would help with that, something like:
```python
info = CaseInfo(report, case)
if info.owner_type == 'location':
    return user_can_access_location_id(domain, user, info.owner_id)
elif info.owner_type == 'user':
    owning_user = CommCareUser.get(info.owner_id)
    return user_can_access_other_user(domain, user, owning_user)
else:
    return False
```
If the case is owned by a group, it shouldn't ever be accessible to restricted users.